vignettes/higher_order.Rmd

```diff
@@ -0,0 +1,312 @@
+---
+title: "Higher Order Functions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Higher Order Functions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This vignette explains some of the more advanced options of `{admiral}` related
+to higher order functions. A higher order function is a function that takes another
+function as input. By introducing these higher order functions, we intend to give
+the user greater power over our derivation functions, whilst trying to negate the
+need for adding additional `{admiral}` functions or arguments, or the user needing
+many separate steps.
+
+The functions covered here are:
+
+* `call_derivation()`: Call a single derivation multiple times with some
+arguments being fixed across iterations and others varying
+* `restrict_derivation()`: Execute a derivation on a subset of the input dataset
+* `slice_derivation()`: The input dataset is split into slices (subsets) and for
```

Comment:
can one only use one function with different arguments per slice or can one call different functions per slice? Would be great to point this out.